libs/client-api/src/v2/README.md

```diff
@@ -0,0 +1,929 @@
+   which update takes precedence:
+   ```rust
+   // In client_api/src/v2/actor.rs
+   last_message_id.rcu(|old| {
+     if rid > **old {
+       Arc::new(rid)
+     } else {
+       old.clone()
+     }
+   });
+   ```
+
+#### Rid in Server-Side Implementation
+
+In the server implementation (`services/appflowy-collaborate`), Rids play a crucial role:
+
+1. **Update Ordering**: The `GroupCommandRunner` in the server uses Rids to process updates in the correct order:
+   ```rust
+   // Pseudocode based on services/appflowy-collaborate/src/group/group_init.rs
+   async fn handle_update(
+     state: &CollabGroupState,
+     origin: &CollabOrigin,
+     update: Vec<u8>,
+     rid: Rid,
+   ) -> Result<(), RTProtocolError> {
+     // Process update
+     // Track the Rid for this update in the server state
+     state.update_last_message_id(rid);
+     // Forward update to other clients with this Rid
+     broadcast_to_subscribers(update, rid);
+   }
+   ```
+
+2. **Update Validation**: The server verifies if an update is already applied using Rid comparison:
+   ```rust
+   // Simplified implementation
+   if received_rid <= last_processed_rid {
+     // Update already processed, can be ignored
+     return Ok(());
+   }
+   ```
+
+3. **Message Forwarding**: When broadcasting updates to other clients, the server includes the original Rid:
+   ```rust
+   // In services/appflowy-collaborate/src/group/group_init.rs
+   let message = BroadcastSync::new(
+     sender,
+     object_id.to_string(),
+     payload,
+     seq_num,
+     original_rid
+   );
+   ```
+
+#### Benefits of Rid-based Ordering
+
+- **Global Ordering**: Provides total ordering of updates across all clients
+- **Idempotency**: Duplicate update detection by tracking last_message_id
+- **Out-of-order Handling**: Later updates are recognized even if they arrive first
+- **Reconnection Efficiency**: Clients request updates newer than their last_message_id
+- **Causality Preservation**: Updates are applied in a causally consistent order
+- **Conflict Resolution Support**: Rid ordering helps resolve conflicts when merged
+
+The Rid mechanism, combined with CRDT state vectors, forms the foundation of AppFlowy's robust synchronization system,
+providing consistent document states even in challenging network conditions.
+
+### Disconnection and Reconnection
+
+When a client disconnects, the system:
+
+- Keeps local changes in memory and continues tracking the state vector
+- Updates client state to `ConnectionStatus::Disconnected`
+- Uses the `ReconnectionManager` to attempt reconnection with exponential backoff
+
+```rust
+async fn handle_disconnect(&self) -> anyhow::Result<()> {
+    let previous_status = self
+        .status_tx
+        .send_replace(ConnectionStatus::Disconnected { reason: None });
+    // ... disconnect logic
+}
+```
+
+The reconnection process is managed by a dedicated `ReconnectionManager` that:
```

Comment:
There's no such thing as `ReconnectionManager` as far as I know (at least not in this repo).

Reply:
I implemented it